meta-iot2000-example/recipes-support/swupdate/swupdate/Makefile

```diff
@@ -0,0 +1,23 @@
+# Makefile for swupdate
+#
+# Copyright (C) 2013 Stefano Babic <sbabic@denx.de>
+#
+# Based on busybox build system:
+#
+# Copyright (C) 1999-2005 by Erik Andersen <andersen@codepoet.org>
+#
+# SPDX-License-Identifier:     GPL-2.0-or-later
+
+lib-y += \
+	 client.o \
+	 progress.o \
+	 hawkbitcfg.o \
+	 sendtohawkbit.o
+
+# # Uncomment the next lines to integrate the compiling/linking of
+# # any .c files placed alongside the above "official" tools in the
+# # tools/ directory into SWUpdate's build system.
+TOOLS_DIR := $(dir $(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST)))
```

Comment:
Hm, what about patching the `Makefile` instead of replacing it? This is less flexible...

Reply:
If the makefile changes, a patch cannot be applied anymore. If the replacement does not work, it has changed. So functionally, it is the same and should not matter regarding flexibility. Just deploying a file is easier than patching it from the recipe's point of view.

Reply:
Okay that's a good point.